cmd/dynkeepalived/dynkeepalived.go

```diff
@@ -42,14 +50,16 @@ func main() {
 				return err
 			}
 
-			return monitor.KeepalivedWatch(args[0], clusterConfigPath, args[1], args[2], apiVip, ingressVip, dnsVip, checkInterval)
+			return monitor.KeepalivedWatch(args[0], clusterConfigPath, args[1], args[2], apiVip, ingressVip, dnsVip, apiPort, lbPort, checkInterval)
 		},
 	}
 	rootCmd.PersistentFlags().StringP("cluster-config", "c", "", "Path to cluster-config ConfigMap to retrieve ControlPlane info")
-	rootCmd.Flags().Duration("check-interval", time.Second*10, "Time between keepalived watch checks")
+	rootCmd.Flags().Duration("check-interval", time.Second*1, "Time between keepalived watch checks")
```

Comment:
the only trouble I can think of is to be sending reload commands to keepalived all too often.

Reply:
I think that this change will break deployment with unicast-keepalived.

Reply:
HAProxy monitor adds the firewall rule after 3*6 Secs and removes it after 2*6 Secs, what's the added value checking rule existence every second?

Reply:
Oops, I missed these comments. The issue with a 10 second check interval is that if haproxy has an issue and needs to remove the firewall rule, it will take 12 seconds for that to happen, and if we're only checking every 10 it may take up to 10 seconds for keepalived to notice that. It nearly doubles the response time to outages (12 to 22 seconds). In the past, keepalived has responded to outages in a couple of seconds at most.

Maybe the longer response time is acceptable. I don't know what our goal for VIP failover is, so I've been trying to keep it as short as possible.

Reply:
Well, the monitor sidecars are used for configuration update, and for this purpose, it's reasonable to have slow detection (interval 6-10 Seconds), with this PR, keepalived-monitor sidecar is involved also in failure detection and of course service recovery so it should detect the failure ASAP. 
I know that you moved this code to monitor sidecar because keepalived container doesn't support accessing to iptable information. 

Do you think that adding a new sidecar container to keepalived that will be responsible for this logic is an option?

If we'll decide to keep this logic in keepalived-monitor I think we should have separate timing (and even code/function if it's possible) for failure detection and config update, there is no added value to check for config update every 1 second.

Reply:
I put it back to 10 seconds. If it ends up being a problem we can revisit.